File: api-pytest-framework/features/steps/auth_steps.py
"""Steps BDD pour auth.feature -- POST /auth."""
from pytest_bdd import when, then
from pages.auth_page import AuthPage


# ── When : POST /auth ─────────────────────────────────────────────────────────

@when('j\'envoie POST /auth avec username "admin" et password "password123"')
def post_auth_valid(ctx):
    ctx["response"] = ctx["auth"].create_token("admin", "password123")


@when('j\'envoie POST /auth avec username "wrong" et password "wrong"')
def post_auth_invalid(ctx):
    ctx["response"] = ctx["auth"].create_token("wrong", "wrong")


@when("j'envoie POST /auth avec un body vide {}")
def post_auth_empty(ctx):
    ctx["response"] = ctx["auth"].post("/auth", {})


@when("j'envoie POST /auth avec une injection SQL dans username")
def post_auth_sql_injection(ctx):
    ctx["response"] = ctx["auth"].create_token("' OR '1'='1", "test")


@when("j'envoie POST /auth avec un payload XSS dans password")
def post_auth_xss(ctx):
    ctx["response"] = ctx["auth"].create_token("admin", '<script>alert("XSS")</script>')


# ── Then : assertions auth ────────────────────────────────────────────────────

@then('la reponse contient un champ "token" non vide')
def check_token_present(ctx):
    body = ctx["response"].json()
    assert "token" in body, f"Champ 'token' absent : {body}"
    assert body["token"], "Le token est vide"


@then("la longueur du token est superieure a 10 caracteres")
def check_token_length(ctx):
    token = ctx["response"].json().get("token", "")
    assert len(token) > 10, f"Token trop court ({len(token)} chars) : {token}"


@then('la reponse contient {"reason": "Bad credentials"}')
def check_bad_credentials(ctx):
    body = ctx["response"].json()
    assert body.get("reason") == "Bad credentials", (
        f"Message d'erreur inattendu : {body}"
    )

File: api-Java-Rest-Assured/src/test/java/com/restfulbooker/steps/AuthSteps.java
package com.restfulbooker.steps;

import com.restfulbooker.context.ScenarioContext;
import io.cucumber.java.en.Then;
import io.cucumber.java.en.When;
import io.restassured.path.json.JsonPath;

import java.util.Map;

import static org.testng.Assert.*;

/**
 * Steps BDD pour auth.feature -- POST /auth.
 * Équivalent Java de features/steps/auth_steps.py.
 */
public class AuthSteps {

    private final ScenarioContext ctx;

    public AuthSteps(ScenarioContext ctx) {
        this.ctx = ctx;
    }

    @When("^j'envoie POST /auth avec username \"admin\" et password \"password123\"$")
    public void postAuthValid() {
        ctx.setResponse(ctx.getAuth().createToken("admin", "password123"));
    }

    @When("^j'envoie POST /auth avec username \"wrong\" et password \"wrong\"$")
    public void postAuthInvalid() {
        ctx.setResponse(ctx.getAuth().createToken("wrong", "wrong"));
    }

    @When("^j'envoie POST /auth avec un body vide \\{\\}$")
    public void postAuthEmpty() {
        ctx.setResponse(ctx.getAuth().createTokenEmpty());
    }

    @When("^j'envoie POST /auth avec une injection SQL dans username$")
    public void postAuthSqlInjection() {
        ctx.setResponse(ctx.getAuth().createToken("' OR '1'='1", "test"));
    }

    @When("^j'envoie POST /auth avec un payload XSS dans password$")
    public void postAuthXss() {
        ctx.setResponse(ctx.getAuth().createToken("admin", "<script>alert(\"XSS\")</script>"));
    }

    // ── Then : assertions auth ────────────────────────────────────────────────

    @Then("^la reponse contient un champ \"token\" non vide$")
    public void checkTokenPresent() {
        JsonPath json = ctx.getResponse().jsonPath();
        String token = json.getString("token");
        assertNotNull(token, "Champ 'token' absent");
        assertFalse(token.isEmpty(), "Le token est vide");
    }

    @Then("^la longueur du token est superieure a 10 caracteres$")
    public void checkTokenLength() {
        String token = ctx.getResponse().jsonPath().getString("token");
        int length = token != null ? token.length() : 0;
        assertTrue(length > 10, "Token trop court (" + length + " chars) : " + token);
    }

    @Then("^la reponse contient \\{\"reason\": \"Bad credentials\"\\}$")
    public void checkBadCredentials() {
        Map<String, Object> body = ctx.getResponse().jsonPath().getMap("$");
        assertEquals(body.get("reason"), "Bad credentials", "Message d'erreur inattendu : " + body);
    }
}
